gnu_dataset/gnats-4.2.0/doc/man/file-pr.man
.\" Copyright (c) 1993, 1999, 2003 Free Software Foundation, Inc.
.\" See section COPYING for conditions for redistribution
.TH file-pr 8 "August 2003" "GNATS 4.2.0" "GNATS Internal Utilities"
.de BP
.sp
.ti -.2i
\(**
..
.SH NAME
file-pr \- files incoming problem reports in GNATS database
.SH SYNOPSIS
.hy 0
.na
.B file\-pr
[
.B \-f \fIfile\fB 
| 
.B \-\-file=\fIfile\fB
]
[
.B \-h 
| 
.B \-\-help
]
.br
[
.B \-V 
| 
.B \-\-version
]
[
.BI -H \ host
|
.BI --host= host
]
.br
[
.BI -P \ port
|
.BI --port= port
]
[
.BI -v \ user
|
.BI --user= user
]
.br
[
.BI -w \ password
|
.BI --passwd= password
]
.br
[
.BI -d \ databasename
|
.BI --database= databasename
]

.ad b
.hy 1
.SH DESCRIPTION
\fBfile-pr\fR files incoming Problem Reports in the GNATS
database, sends acknowledgments to approriate parties and logs
database activity.
.P
\fBfile-pr\fR assigns the Problem Report an identification number and
then files it in the GNATS database either under the category
specified in the Category field of the PR, or under the default
category if the Category field is nonexistent or contains an invalid
category name.
.P
For the default GNATS configuration, the person responsible for that
category and the person responsible for the submitter site where the
PR originated receive a copy of the PR in its entirety.  Optionally,
the originator of the PR receives an acknowledgment that the PR
arrived and was filed
.SH OPTIONS
.TP
.B \-f \fIfile\fB, \-\-filename=\fIfile\fB
Accept
.I file
as input; if this option is not present, standard input is used.
.TP
.B \-h, \-\-help
Display usage summary for
.BR file\-pr .
.TP
.B \-V, \-\-version
Display version number for
.BR file\-pr .
.TP
.B -d\fR,\fB --database
Specifies the database to which the PR is to be submitted.  If no
database is specified, the database named \fIdefault\fR is assumed.
This option overrides the database specified in the \fBGNATSDB\fR
environment variable.
.SS
GNATS network options:
.TP
.B -H\fR,\fB --host
Hostname of the GNATS server.
.TP
.B -P\fR,\fB --port
The port that the GNATS server runs on.
.TP
.B -v\fR,\fB --username
Username used to log into the GNATS server.
.TP
.B -w\fR,\fB --passwd
Password used to log into the GNATS server.
.SH ENVIRONMENT VARIABLES
The \fBGNATSDB\fR environment variable is used to determine which database to
use.  For a local database, it contains the name of the database to access.
.P
For network access via \fIgnatsd\fR, it contains a colon-separated list
of strings that describe the remote database, in the form
.TP
\fIserver\fR:\fIport\fR:\fIdatabasename\fR:\fIusername\fR:\fIpassword\fR
.P
Any of the fields may be omitted, but at least one colon must appear;
otherwise, the value is assumed to be the name of a local database.
.P
If \fBGNATSDB\fR is not set, it is assumed that the database is local
and that its name is \fIdefault\fR.
.SH "SEE ALSO"
.I Keeping Track: Managing Messages With GNATS
(also installed as the GNU Info file
.BR gnats.info )
.LP
.BR databases (5),
.BR dbconfig (5),
.BR delete-pr (8),
.BR edit-pr (1)
.BR file-pr (8),
.BR gen-index (8),
.BR gnats (7),
.BR gnatsd (8),
.BR mkcat (8),
.BR mkdb (8),
.BR pr-edit (8),
.BR query-pr (1),
.BR queue-pr (8),
.BR send-pr (1).
.SH COPYING
Copyright (c) 1993, 1999, 2003, Free Software Foundation, Inc.
.PP
Permission is granted to make and distribute verbatim copies of
this manual provided the copyright notice and this permission notice
are preserved on all copies.
.PP
Permission is granted to copy and distribute modified versions of this
manual under the conditions for verbatim copying, provided that the
entire resulting derived work is distributed under the terms of a
permission notice identical to this one.
.PP
Permission is granted to copy and distribute translations of this
manual into another language, under the above conditions for modified
versions, except that this permission notice may be included in
translations approved by the Free Software Foundation instead of in
the original English.


gnu_dataset/gnats-4.2.0/gnats/file-pr.sh
#!/bin/sh
# Program to submit problem reports for GNATS.
# Copyright (C) 1999,2007 Free Software Foundation, Inc.
# Contributed by Bob Manson (manson@juniper.net).
#
# This file is part of GNU GNATS.
#
# GNU GNATS is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 3, or (at your option)
# any later version.
#
# GNU GNATS is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with GNU GNATS; see the file COPYING.  If not, see
# <http://www.gnu.org/licenses/>.

debug_print=false # or echo to get output.

PATH="@pkglibexecdir@:@sbindir@:@bindir@:/bin:/usr/bin:${PATH}"
export PATH

EDIT_USER=
EDIT_PASSWD=
version="@VERSION@"

usage="
Usage:  file-pr [-V|--version] [-h|--help] [-d|--database database_name]
                [-H|--host hostname] [-P|--port port_number]
                [-v|--user userid]   [-w|--passwd password]
                [-f|--filename inputfile]
"

# Parse command line.  We don't really need to do this, but that's ok.

while [ $# -gt 0 ]; do
   case "$1" in
     -V|--version|--ve*)
       echo "$version"; exit 0
       ;;

     -d | --database)
       if [ $# -eq 1 ]; then echo "$usage"; exit 1; fi
       shift ; GNATS_DB="--database=$1" ;;
     -d=* | --database=*) GNATS_DB="$1" ;;

     -D | --debug)
       shift ;;

     -H | --host)
       if [ $# -eq 1 ]; then echo "$usage"; exit 1; fi
       shift ; GNATS_HOST="--host=$1" ;;
     -H=* | --host=*) GNATS_HOST="$1" ;;

     -P | --port)
       if [ $# -eq 1 ]; then echo "$usage"; exit 1; fi
       shift ; GNATS_PORT="--port=$1" ;;
     -P=* | --port=*) GNATS_PORT="$1" ;;

     -v | --user)
       if [ $# -eq 1 ]; then echo "$usage"; exit 1; fi
       shift ; EDIT_USER="--user=$1" ;;
     -v=* | --user=*) EDIT_USER="$1" ;;

     -w | --passwd)
       if [ $# -eq 1 ]; then echo "$usage"; exit 1; fi
       shift ; EDIT_PASSWD="--passwd=$1" ;;
     -w=* | --passwd=*) EDIT_PASSWD="$1" ;;
     -f|--filename*)
       if [ $# -eq 1 ]; then echo "$usage"; exit 1; fi
       shift ; FILENAME="$1" ;;
     -h|--help*)
       echo "$usage"; exit 0
       ;;
     *)
       echo "$usage"; exit 1
       ;;
   esac
   shift
done

# set command here to always pass host and port, and directory if supplied
PR_EDIT="pr-edit ${GNATS_HOST} ${GNATS_PORT} ${EDIT_USER} ${EDIT_PASSWD} ${GNATS_DB}"

if [ "$FILENAME" != "" ]; then
    PR_EDIT="$PR_EDIT -f $FILENAME"
fi

$PR_EDIT --submit
